SpatialGDK/Source/SpatialGDKServices/Private/SSpatialOutputLog.cpp

```diff
@@ -303,74 +302,61 @@ void SSpatialOutputLog::FormatAndPrintRawErrorLine(const FString& LogLine)
-	// FString LogMessage = LogMatcher.GetCaptureGroup(2);
-	// FString LogCategory = LogMatcher.GetCaptureGroup(4);
-
-	//// For worker logs 'WorkerLogMessageHandler' we use the worker name as the category. The worker name can be found in the msg.
-	//// msg=[WORKER_NAME:WORKER_TYPE] ... e.g. msg=[UnrealWorkerF5C56488482FEDC37B10E382770067E3:UnrealWorker]
-	// if (LogCategory == TEXT("WorkerLogMessageHandler") || LogCategory == TEXT("Runtime"))
-	//{
-	//	const FRegexPattern WorkerLogPattern = FRegexPattern(TEXT("\\[([^:]*):([^\\]]*)\\] (.*)"));
-	//	FRegexMatcher WorkerLogMatcher(WorkerLogPattern, LogMessage);
-
-	//	if (WorkerLogMatcher.FindNext())
-	//	{
-	//		LogCategory = WorkerLogMatcher.GetCaptureGroup(1);		  // Worker Name
-	//		FString WorkerType = WorkerLogMatcher.GetCaptureGroup(2); // Worker Type
-
-	//		if (LogCategory.StartsWith(WorkerType))
-	//		{
-	//			// We shorten the category name to make it more human readable. e.g. UnrealWorkerF5C56
-	//			LogCategory = LogCategory.Left(WorkerType.Len() + 5);
-	//		}
-
-	//		LogMessage = WorkerLogMatcher.GetCaptureGroup(3);
-	//	}
-	//}
-
-	// ELogVerbosity::Type LogVerbosity = ELogVerbosity::Display;
-
-	// if (LogLevelText.Contains(TEXT("error")))
-	//{
-	//	LogVerbosity = ELogVerbosity::Error;
-	//}
-	// else if (LogLevelText.Contains(TEXT("warn")))
-	//{
-	//	LogVerbosity = ELogVerbosity::Warning;
-	//}
-	// else if (LogLevelText.Contains(TEXT("debug")))
-	//{
-	//	LogVerbosity = ELogVerbosity::Verbose;
-	//}
-	// else if (LogLevelText.Contains(TEXT("verbose")))
-	//{
-	//	LogVerbosity = ELogVerbosity::Verbose;
-	//}
-	// else
-	//{
-	//	LogVerbosity = ELogVerbosity::Log;
-	//}
+	 const FRegexPattern LogPattern = FRegexPattern(TEXT("\\[(\\w*)\\] \\[(\\w*)\\] (.*)"));
+	 FRegexMatcher LogMatcher(LogPattern, LogLine);
 
-	// Serialization must be done on the game thread.
-	// AsyncTask(ENamedThreads::GameThread, [this, LogMessage, LogVerbosity, LogCategory] {
-	//	Serialize(*LogMessage, LogVerbosity, FName(*LogCategory));
-	//});
+	 if (!LogMatcher.FindNext())
+	{
+		// If this log line did not match the log line regex then it is an error line which is parsed differently.
+		FormatAndPrintRawErrorLine(LogLine);
+		//UE_LOG(LogSpatialOutputLog, Error, TEXT("Failed to parse log line: %s"), *LogLine);
+		return;
+	}
+	 FString LogCategory = LogMatcher.GetCaptureGroup(1);
+	 FString LogLevelText = LogMatcher.GetCaptureGroup(2);
+	 FString LogMessage = LogMatcher.GetCaptureGroup(3);
+
+	 const FRegexPattern LogMessagePattern = FRegexPattern(TEXT("\\[(.*)\\] (.*)"));
+	 FRegexMatcher LogMatcherMessage(LogMessagePattern, LogMessage);
+
+	 if (LogMatcherMessage.FindNext())
+	 {
+		 FString WorkerType = "UnrealWorker";
+		 FString LogMessageCategory = LogMatcherMessage.GetCaptureGroup(1);
+		 LogMessage = LogMatcherMessage.GetCaptureGroup(2);
+		 LogCategory = LogMessageCategory.Left(WorkerType.Len() + 5);
+	 }
+	 else
+	 {
+		 LogCategory = "Runtime";
```

Comment:
might want to have a short comment here as to why it's "Runtime" if we can't find a match

Reply:
And use 
LogCategory = TEXT("Runtime");
for FString